core-to-plc/test/IllTyped.hs

```diff
@@ -39,3 +47,28 @@ ptreeConstruct = plc (Two (Two (One ((1,2),(3,4)))) :: B Int)
 -- TODO: replace this with 'first' when we have working recursive functions
 ptreeMatch :: PlcCode
 ptreeMatch = plc (\(t::B Int) -> case t of { One a -> a; Two _ -> 2; })
+
+fib :: PlcCode
+-- not using case to avoid literal cases
+fib = plc (let fib :: Int -> Int
+               fib n = if n == 0 then 0 else if n == 1 then 1 else fib(n-1) + fib(n-2)
+           in fib)
+
+sumDirect :: PlcCode
+sumDirect = plc (let sum :: [Int] -> Int
+                     sum []     = 0
+                     sum (x:xs) = x + sum xs
+                 in sum)
```

Comment:
Why does this required mutual recursion?

Reply:
It doesn't! But it uses the same machinery. I meant to add an even/odd example, I'll do that.

Reply:
Added.